### PR TITLE
Prometheus: Fix passing query timeout to upstream queries

### PR DESCRIPTION
**Pull request by itsmylife**
**What is this feature?**

In Prometheus data source we have an additional "query timeout" value in the settings. This was being used to limit how long a query will be evaluated. See the docs: https://prometheus.io/docs/prometheus/latest/querying/api/#range-queries
image

During the backend transformation of the data source this was missed to be included to the code. We already have "timeout" configuration. But this is being used to control how long after a query going to grafana backend will be canceled. 
image

So in this PR we put back that upstream query timeout value in place. 

**Why do we need this feature?**

For more explicit query timeout setting.

**Who is this feature for?**

Prometheus users

### CODE REVIEW

File: pkg/promlib/client/client.go [hunk 1]
```diff
@@ -24,22 +24,29 @@ type doer interface {
...
+	baseUrl      string
+	queryTimeout string
 }
 
-func NewClient(d doer, method, baseUrl string) *Client {
-	return &Client{doer: d, method: method, baseUrl: baseUrl}
+const defaultQueryTimeout = "60s"
+
+func NewClient(d doer, method, baseUrl, queryTimeout string) *Client {
+	if queryTimeout != "" {
```
Comment (s4kh) [inline pkg/promlib/client/client.go:36]: Why are we overwriting the passed `queryTimeout`? Should not this be `queryTimeOut == ""`?

Comment (marefr) [reply pkg/promlib/client/client.go:36]: Yes, you're probably right. Question is if it should default to `60s` or not set/passed when empty. Might be a breaking change (or unexpected behavior) for those that haven't configured either timeout or query timeout.

**Review by marefr — COMMENTED** [review]
Looks good in general with some minor changes. As an alternative you could implement this in https://github.com/grafana/grafana/blob/main/pkg/promlib/middleware/custom_query_params.go, but maybe it shouldn't apply to all outgoing prometheus requests?

Might want to test this in https://github.com/grafana/grafana/blob/main/pkg/tests/api/prometheus/prometheus_test.go

File: pkg/promlib/client/client.go (hunk 1 above)
Comment (itsmylife) [reply pkg/promlib/client/client.go:36]: @s4kh @marefr  You are right it is a mistake. Must be `==`.
@marefr That's a good point. if that's not specified we shouldn't use it.

**Comment by itsmylife** [comment]
@marefr I think `custom_query_params.go` middleware applies those params to all requests. I want to apply only for `instant` and `range` query requests. That's why I implemented that way.

**Review by marefr — APPROVED** [review]

**Pull request merged**